src/components/TrackingConsent.vue

```diff
@@ -0,0 +1,467 @@
+            return;
+        }
+        const geoIpInfo = await geoIpResponse.json();
+        if (geoIpInfo.continent !== 'EU' && !['NO', 'LI', 'IS'].includes(geoIpInfo.country)) { /* EU / EEA */
+            this.uiRequired = false;
+            this._initMatomo();
+            return;
+        }
+
+        this.uiRequired = true;
+    }
+
+    private _onVisibilityChange() {
+        if (document.visibilityState === 'visible') {
+            this._checkUiRequired();
+        }
+    }
+
+    private _setConsent(consent: Consent) {
+        TrackingConsent._setCookie(
+            TrackingConsent.COOKIE_STORAGE_KEY,
+            JSON.stringify(consent),
+            this.cookieOptions,
+        );
+        this.uiRequired = false;
+
+        /* update other TrackingConsent instances */
+        TrackingConsent._instances.forEach((instance) => {
+            instance.uiRequired = false;
+        });
+    }
+
+    private _initMatomo() {
+        /* Check whether matomo is already initialized */
+        if (document.querySelector('script[matomo]')) {
+            console.warn('TrackingConsent: Matomo already initialized.');
+            return;
+        }
+
+        /* set mtm startTime */
+        TrackingConsent._mtm.push({
+            'mtm.startTime': window.startTime,
+            'event': 'mtm.Start',
+        });
+
+        /* Get referrer from localstorage */
+        const referrer = localStorage.getItem('referrer');
+        if (referrer) {
+            TrackingConsent._paq.push(['setReferrerUrl', decodeURIComponent(referrer)]);
+            localStorage.removeItem('referrer');
+        }
+
+        /* set setTrackerUrl to default value if not set */
+        if (!this.options.setTrackerUrl) {
+            this.options.setTrackerUrl = TrackingConsent.DEFAULT_TRACKER_URL;
+        }
+
+        /* Cycle through options and set them */
+        Object.keys(this.options).forEach((k) => {
+            const option = this.options[k];
+
+            TrackingConsent._paq.push(option === null ? [k] : [k, option]);
+        });
+
+        /* append script */
+        const script = document.createElement('script');
+        const nextScript = document.getElementsByTagName('script')[0];
+        script.type = 'text/javascript';
+        script.async = script.defer = true;
+        script.src = this.trackingScriptUrl;
+        script.setAttribute('matomo', '');
+        nextScript.parentNode.insertBefore(script, nextScript);
+    }
+}
+
+namespace TrackingConsent { // tslint:disable-line:no-namespace
+    export enum Themes {
+        LIGHT = 'light',
+        DARK = 'dark',
+    }
+
+    export const LOCALSTORAGE_KEYS = [
+        'tracking-consent',
+        'tracking-consensus',
+    ];
+    export const COOKIE_STORAGE_KEY = 'tracking-consent';
+
+    export const DEFAULT_COOKIE_DOMAIN = document.location.hostname;
+    export const DEFAULT_COOKIE_EXPIRATION_DAYS = 365 * 20;
```

Comment:
20 years? :laughing:

Reply:
Actually let me change it to 10 years, because 20 years won't work on 32bits systems.